public/js/components/Expressions.js

```diff
@@ -3,66 +3,111 @@ const { connect } = require("react-redux");
   propTypes: {
-    pauseInfo: ImPropTypes.map,
     expressions: ImPropTypes.list,
     addExpression: PropTypes.func,
     updateExpression: PropTypes.func,
-    evaluateExpression: PropTypes.func,
-    loadObjectProperties: PropTypes.func,
-    command: PropTypes.func
+    deleteExpression: PropTypes.func
   },
 
   displayName: "Expressions",
 
   addExpression(e) {
+    const { addExpression } = this.props;
     if (e.key === "Enter") {
       const expression = {
         input: e.target.value
       };
       if (e.target.id) {
         expression.id = e.target.id.split("-").pop();
       }
-      this.props.addExpression(expression);
+      e.target.value = "";
+      addExpression(expression);
     }
   },
 
   updateExpression(e) {
+    const { updateExpression } = this.props;
+    const id = e.target.id.split("-").pop();
     const expression = {
-      id: e.target.id.split("-").pop(),
-      input: e.target.textContent.split(" --> ")[0]
+      id,
+      input: e.target.textContent
     };
-    this.props.updateExpression(expression);
+    updateExpression(expression);
+    setTimeout(() => {
+      document.getElementById("expressionInputId-" + id).focus();
+    }, 1);
+  },
+
+  renderExpressionValue(value) {
+    if (!value) {
+      return;
+    }
+    if (value.exception) {
+      return;
+    }
+    if (typeof value.result !== "object") {
+      return value.result;
+    }
+    if (typeof value.result === "object") {
+      return Rep({ object: value.result });
+    }
+  },
+
+  deleteExpression(expression, e) {
+    e.stopPropagation();
+    const { deleteExpression } = this.props;
+    deleteExpression(expression);
   },
 
   renderExpression(expression) {
     return dom.div(
       { className: "expression-container",
-        key: expression.id },
+      key: expression.id },
       expression.updating ?
-        dom.input(
-          { type: "text",
-            className: "input-expression",
-            id: "expressionInput-" + expression.id,
-            placeholder: "Add watch Expression",
-            onKeyPress: this.addExpression,
-            defaultValue: expression.input }
+        dom.span(
+          { className: "expression-input-container" },
+          dom.input(
+            { type: "text",
+              className: "input-expression",
+              id: "expressionInputId-" + expression.id,
+              onKeyPress: this.addExpression,
+              defaultValue: expression.input }
+          )
         ) :
         dom.span(
-          { key: expression.id,
-            id: "expressionOutput-" + expression.id,
+          { className: "expression-output-container",
+            key: expression.id,
             onClick: this.updateExpression },
-          expression.input + " --> " +
-          JSON.stringify(expression.value || "Not Paused")
+          dom.span(
+            { className: "expression-input",
+              id: "expressionOutputId-" + expression.id },
```

Comment:
using ID in this way is a code smell. The id could be a data attribute `data-id=expression.value` and accessed easily